import sys
from PyQt5 import QtCore, QtGui, QtWidgets
from beni.controller.ControlloreListaBeni import *
import requests


class Ui_VistaBene(object):

    def __init__(self, utente_attivo, url):
        super(Ui_VistaBene, self).__init__()
        self.controller = ControlloreListaBeni()
        self.utente_attivo = utente_attivo
        self.image_url = url
    def setupUi(self, VistaBene):
        VistaBene.setObjectName("VistaBene")
        VistaBene.resize(800, 600)
        self.centralwidget = QtWidgets.QWidget(VistaBene)
        self.centralwidget.setObjectName("centralwidget")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(10, 190, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        self.label_6.setGeometry(QtCore.QRect(10, 330, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(10, 120, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.lineEdit_4 = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_4.setGeometry(QtCore.QRect(10, 210, 401, 51))
        self.lineEdit_4.setText("")
        self.lineEdit_4.setMaxLength(5000)
        self.lineEdit_4.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.lineEdit_4.setObjectName("lineEdit_4")
        self.lineEdit_7 = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_7.setGeometry(QtCore.QRect(10, 400, 61, 21))
        self.lineEdit_7.setReadOnly(True)
        self.lineEdit_7.setObjectName("lineEdit_7")
        self.label_5 = QtWidgets.QLabel(self.centralwidget)
        self.label_5.setGeometry(QtCore.QRect(10, 270, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(10, 60, 71, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.lineEdit_2 = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_2.setGeometry(QtCore.QRect(610, 80, 171, 21))
        self.lineEdit_2.setObjectName("lineEdit_2")
        self.lineEdit = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit.setGeometry(QtCore.QRect(10, 80, 171, 21))
        self.lineEdit.setObjectName("lineEdit")
        self.comboBox = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox.setGeometry(QtCore.QRect(10, 150, 141, 22))
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.checkBox = QtWidgets.QCheckBox(self.centralwidget)
        self.checkBox.setGeometry(QtCore.QRect(10, 300, 141, 21))
        self.checkBox.setObjectName("checkBox")
        self.label_8 = QtWidgets.QLabel(self.centralwidget)
        self.label_8.setGeometry(QtCore.QRect(10, 430, 161, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_8.setFont(font)
        self.label_8.setObjectName("label_8")
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        self.label_7.setGeometry(QtCore.QRect(10, 380, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.label_9 = QtWidgets.QLabel(self.centralwidget)
        self.label_9.setGeometry(QtCore.QRect(350, 0, 291, 31))
        font = QtGui.QFont()
        font.setFamily("Times New Roman")
        font.setPointSize(20)
        self.label_9.setFont(font)
        self.label_9.setObjectName("label_9")
        self.checkBox_2 = QtWidgets.QCheckBox(self.centralwidget)
        self.checkBox_2.setGeometry(QtCore.QRect(10, 350, 141, 21))
        self.checkBox_2.setObjectName("checkBox_2")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(630, 50, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.lineEdit_8 = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_8.setGeometry(QtCore.QRect(10, 460, 91, 21))
        self.lineEdit_8.setObjectName("lineEdit_8")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setEnabled(True)
        self.pushButton.setGeometry(QtCore.QRect(670, 490, 121, 51))
        self.pushButton.setObjectName("pushButton")
        self.pushButton_2 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_2.setEnabled(True)
        self.pushButton_2.setGeometry(QtCore.QRect(530, 490, 121, 51))
        self.pushButton_2.setObjectName("pushButton_2")
        VistaBene.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(VistaBene)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 800, 22))
        self.menubar.setObjectName("menubar")
        VistaBene.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(VistaBene)
        self.statusbar.setObjectName("statusbar")
        VistaBene.setStatusBar(self.statusbar)
        self.pushButton_3 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_3.setGeometry(QtCore.QRect(390, 490, 121, 51))
        self.pushButton_3.setObjectName("pushButton_3")

        response = requests.get(self.image_url)
        image_data = response.content

        pixmap = QtGui.QPixmap()
        pixmap.loadFromData(image_data)

        label = QtWidgets.QLabel(self.centralwidget)
        label.setGeometry(QtCore.QRect(610, 110, 171, 171))
        label.setPixmap(pixmap)
        label.setScaledContents(True)
        label.setObjectName("imageLabel")
        label.show()

        self.retranslateUi(VistaBene)
        QtCore.QMetaObject.connectSlotsByName(VistaBene)

        self.retranslateUi(VistaBene)
        QtCore.QMetaObject.connectSlotsByName(VistaBene)

    def retranslateUi(self, VistaBene):
        _translate = QtCore.QCoreApplication.translate
        VistaBene.setWindowTitle(_translate("VistaBene", "Vista bene"))
        self.label_4.setText(_translate("VistaBene", "Descrizione"))
        self.label_6.setText(_translate("VistaBene", "Stato area"))
        self.label_3.setText(_translate("VistaBene", "Area"))
        self.label_5.setText(_translate("VistaBene", "Stato bene"))
        self.label.setText(_translate("VistaBene", "Nome"))
        self.comboBox.setItemText(0, _translate("VistaBene", "Area Geologica"))
        self.comboBox.setItemText(1, _translate("VistaBene", "Area Zoologica"))
        self.comboBox.setItemText(2, _translate("VistaBene", "Area Paleontologica"))
        self.comboBox.setItemText(3, _translate("VistaBene", "Area esposizione temporanee"))
        self.comboBox.setItemText(4, _translate("VistaBene", "Science room"))
        self.checkBox.setText(_translate("VistaBene", "CheckBox"))
        self.label_8.setText(_translate("VistaBene", "Data di aggiunta"))
        self.label_7.setText(_translate("VistaBene", "ID"))
        self.label_9.setText(_translate("VistaBene", "Bene"))
        self.checkBox_2.setText(_translate("VistaBene", "CheckBox"))
        self.label_2.setText(_translate("VistaBene", "URL immagine"))
        self.pushButton.setText(_translate("VistaBene", "Conferma"))
        self.pushButton_2.setText(_translate("VistaBene", "Aggiorna bene"))
        self.pushButton_3.setText(_translate("VistaBene", "Elimina bene"))


def show_vista_bene(utente_attivo, url):
    ui = Ui_VistaBene(utente_attivo, url)
    ui.setupUi(VistaBene)
    VistaBene.show()
    return ui


app = QtWidgets.QApplication(sys.argv)
VistaBene = QtWidgets.QMainWindow()

